cmd/juju/crossmodel/show.go

```diff
@@ -102,24 +104,44 @@ func (c *showCommand) Run(ctx *cmd.Context) (err error) {
 	}
 	url, err := crossmodel.ParseOfferURL(c.url)
 	if err != nil {
-		currentModel, err := c.ModelIdentifier()
-		if err != nil {
+		if !names.IsValidApplication(c.url) {
 			return errors.Trace(err)
 		}
-		url, err = makeURLFromCurrentModel(c.url, controllerName, currentModel)
-		if err != nil {
-			return errors.Trace(err)
+		url = &crossmodel.OfferURL{
+			ApplicationName: c.url,
 		}
 	}
-	if url.Source != "" {
-		controllerName = url.Source
-	}
 	accountDetails, err := c.CurrentAccountDetails()
 	if err != nil {
 		return err
 	}
 	loggedInUser := accountDetails.User
 
+	if url.ModelName == "" {
+		currentModel, err := c.ModelIdentifier()
```

Comment:
Could we add a helper method to the command base `ModelNameWithQualifier() (string, string, error)`?
It would allow us to unit test it independently of this.

Reply:
yup